packages/eslint-plugin/src/rules/no-restricted-imports.ts

```diff
@@ -172,13 +172,13 @@ const schema: JSONSchema4AnyOfSchema = {
 function isObjectOfPaths(
   obj: unknown,
 ): obj is { paths: ArrayOfStringOrObject } {
-  return Object.prototype.hasOwnProperty.call(obj, 'paths');
+  return !!obj && Object.hasOwn(obj, 'paths');
```

Comment:
`Object.hasOwn`, unlike `Object.prototype.hasOwnProperty`, throws an error when the target object is `null` or `undefined`.